storage: Store.Capacity: panic: ... makeslice: cap out of range

The [`clearrange/checks=false`](https://teamcity.cockroachdb.com/viewLog.html?buildId=1518416&buildTypeId=Cockroach_Nightlies_WorkloadNightly&tab=testsInfo) roachtest failed last night. It passed the import phase, and in the `drop bank table` phase one of the nodes died unexpectedly with:

```
panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range

goroutine 10313 [running]:
panic(0x3b5a260, 0x48ac210)
	/usr/local/go/src/runtime/panic.go:565 +0x2c5 fp=0xc003148ab0 sp=0xc003148a20 pc=0x78d165
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000b9a000, 0x49694e0, 0xc0032443c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x121 fp=0xc003148b10 sp=0xc003148ab0 pc=0x12d0481
runtime.call32(0x0, 0x4191348, 0xc0073557f0, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:519 +0x3b fp=0xc003148b40 sp=0xc003148b10 pc=0x7bb1db
panic(0x3b5a260, 0x48ac210)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5 fp=0xc003148bd0 sp=0xc003148b40 pc=0x78d055
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000b9a000, 0x49694e0, 0xc0032443c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x121 fp=0xc003148c30 sp=0xc003148bd0 pc=0x12d0481
runtime.call32(0x0, 0x4191348, 0xc003d9efd0, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:519 +0x3b fp=0xc003148c60 sp=0xc003148c30 pc=0x7bb1db
panic(0x3b5a260, 0x48ac210)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5 fp=0xc003148cf0 sp=0xc003148c60 pc=0x78d055
runtime.panicmakeslicecap(...)
	/usr/local/go/src/runtime/slice.go:31
runtime.makeslice(0x3a18180, 0x0, 0xffffffffffffffff, 0xd8106d)
	/usr/local/go/src/runtime/slice.go:46 +0xf2 fp=0xc003148d20 sp=0xc003148cf0 pc=0x7a3db2
github.com/cockroachdb/cockroach/pkg/storage.(*Store).Capacity(0xc00080aa00, 0xc00348f100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2103 +0x1b8 fp=0xc003149110 sp=0xc003148d20 pc=0x1c43278
github.com/cockroachdb/cockroach/pkg/storage.(*Store).Descriptor(0xc00080aa00, 0xc000783700, 0xc000000007, 0xc000367e68, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2176 +0x4f fp=0xc003149290 sp=0xc003149110 pc=0x1c4390f
github.com/cockroachdb/cockroach/pkg/server/status.(*MetricsRecorder).GenerateNodeStatus(0xc000488780, 0x49694e0, 0xc0032443c0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/status/recorder.go:462 +0x776 fp=0xc003149720 sp=0xc003149290 pc=0x2f083a6
github.com/cockroachdb/cockroach/pkg/server.(*Node).writeNodeStatus.func1(0x49694e0, 0xc0032443c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:802 +0x80 fp=0xc003149de8 sp=0xc003149720 pc=0x2f93020
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0xc000b9a000, 0x49694e0, 0xc0032443c0, 0x40386d1, 0x1a, 0xc00348fe70, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:283 +0xe7 fp=0xc003149e20 sp=0xc003149de8 pc=0x12d0b47
github.com/cockroachdb/cockroach/pkg/server.(*Node).writeNodeStatus(0xc000103400, 0x49694e0, 0xc0032443c0, 0x4a817c800, 0x1, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:801 +0xa8 fp=0xc003149ea0 sp=0xc003149e20 pc=0x2f69d18
github.com/cockroachdb/cockroach/pkg/server.(*Node).startWriteNodeStatus.func1(0x49694e0, 0xc0032443c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:787 +0x144 fp=0xc003149f78 sp=0xc003149ea0 pc=0x2f92eb4
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc006371980, 0xc000b9a000, 0xc009a91f80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0xfb fp=0xc003149fc8 sp=0xc003149f78 pc=0x12d261b
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc003149fd0 sp=0xc003149fc8 pc=0x7bcef1
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xa8
```

Probably a simple error going on there. I'll take a quick look, though I'm happy for someone else to jump on this as well.

--- a/pkg/storage/metrics.go
+++ b/pkg/storage/metrics.go
@@ -968,7 +968,7 @@ type StoreMetrics struct {
 	registry *metric.Registry
 
 	// Replica metrics.
-	ReplicaCount                  *metric.Gauge // Does not include reserved replicas.
+	ReplicaCount                  *metric.Gauge // Does not include uninitialized or reserved replicas.
 	ReservedReplicaCount          *metric.Gauge
 	RaftLeaderCount               *metric.Gauge
 	RaftLeaderNotLeaseHolderCount *metric.Gauge

--- a/pkg/storage/store_remove_replica.go
+++ b/pkg/storage/store_remove_replica.go
@@ -245,8 +245,6 @@ func (s *Store) removeUninitializedReplicaRaftMuLocked(
 		log.Fatalf(ctx, "uninitialized replica %v was unexpectedly replaced", existing)
 	}
 
-	s.metrics.ReplicaCount.Dec(1)
-
 	// Only an uninitialized replica can have a placeholder since, by
 	// definition, an initialized replica will be present in the
 	// replicasByKey map. While the replica will usually consume the

